lib/filter.js

```diff
@@ -1,15 +1,32 @@
 import {filterDirective} from 'smart-table-core';
 import mixin from './mixin';
 
-const mapPropsToConf = ({smartTable, stFilter, stFilterType = 'string', stFilterOperator = 'includes'}) => ({
+const mapPropsToConf = ({smartTable, stFilter, stFilterType, stFilterOperator}) => ({
   table: smartTable,
   pointer: stFilter,
   type: stFilterType,
   operator: stFilterOperator
 });
 
 export default Object.assign({
-    props: ['smartTable', 'stFilter', 'stFilterType', 'stFilterOperator']
+    props: {
+      smartTable: {
+        type: Object,
+        required: true,
+      },
+      stFilter: {
+        type: String,
+        required: true,
+      },
+      stFilterOperator: {
+        type: String,
```

Comment:
Can this be a function instead? In https://github.com/smart-table/smart-table-filter/commit/8cfe5ebb962701fd4195621ac876cdd91fe66325#diff-4f0c9b7aee91c4fefb317794d8a1e941R65, allow operator to be a custom function or accept a new option to override `operators[operator]` function.